src/content/import_service.cc

```diff
@@ -570,31 +607,51 @@ std::pair<int, bool> ImportService::addContainerTree(int parentContainerId, cons
             for (auto&& [key, val] : configLayoutMapping) {
                 tree = std::regex_replace(tree, std::regex(key), val);
             }
+            auto dirKeys = config->getArrayOption(CFG_IMPORT_VIRTUAL_DIRECTORY_KEYS);
+            auto dirKeyValues = std::vector<std::string>();
+            for (auto&& field : dirKeys) {
+                auto metaField = MetadataHandler::remapMetaDataField(field);
+                auto keyValue = item->getMetaData(metaField);
+                if (!keyValue.empty())
+                    dirKeyValues.push_back(keyValue);
+            }
+            if (!dirKeyValues.empty()) {
+                subTree = fmt::format("{}@{}", tree, fmt::join(dirKeyValues, "@"));
+            } else {
+                subTree = tree;
+            }
+            auto cont = database->findObjectByPath(subTree, DbFileType::Virtual);
+            if (cont && cont->isContainer())
+                containerMap[subTree] = std::dynamic_pointer_cast<CdsContainer>(cont);
+            else
+                containerMap[subTree] = nullptr;
```

Comment:
I'm not able to follow all changes in this commit, but I guess this is the key part that creates a unique ID within the Albums virtual container...
Considering the changes in "src/config/config_definition.cc", it seems that the album artist and the year are considered and therefore it will probably solve Issue #1440.
Looking forward to see this "in action".
Is this part only of the "grb" import mode or also the "traditional"?

Reply:
It will be available for both. I'm still testing in real life.